src/ethereum/EthereumNetwork.ts

```diff
@@ -506,30 +507,18 @@ export class EthereumNetwork {
       params
     }
 
-    let addOnUrl = ''
-    if (url.includes('infura')) {
-      const { infuraProjectId } = this.ethEngine.initOptions
-      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-      if (!infuraProjectId || infuraProjectId.length < 6) {
-        throw new Error('Need Infura Project ID')
-      }
-      addOnUrl = `/${infuraProjectId}`
-    } else if (url.includes('alchemyapi')) {
-      const { alchemyApiKey } = this.ethEngine.initOptions
-      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-      if (!alchemyApiKey || alchemyApiKey.length < 6) {
-        throw new Error('Need Alchemy API key')
-      }
-      addOnUrl = `/v2/-${alchemyApiKey}`
-    } else if (url.includes('quiknode')) {
-      const { quiknodeApiKey } = this.ethEngine.initOptions
-      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-      if (!quiknodeApiKey || quiknodeApiKey.length < 6) {
-        throw new Error('Need Quiknode API key')
+    const regex = /{{(.*?)}}/g
+    const match = regex.exec(url)
+    if (match != null) {
+      const key = match[1]
+      const cleanKey = asEthereumInitKeys(key)
+      const apiKey = this.ethEngine.initOptions[cleanKey]
+      if (typeof apiKey === 'string') {
+        url = url.replace(match[0], apiKey)
+      } else {
+        throw new Error('Incorrect apikey type for RPC')
```

Comment:
If the `apiKey` is one of the values in `initOptions` of `cleanKey`, and if `cleanKey` is a key in the `initOptions`, then the error message should state something like:

```js
`Missing ${apiKey} in 'initOptions' for ${pluginId}`
```

Reply:
apiKey can also be an array, that's what this clause was meant to catch. But i'll add another for the case of `null`